src/controllers/commentController.js

```diff
@@ -0,0 +1,74 @@
+import { createComment, deleteCommentById, getAllComments } from '../services/commentServices';
+import statusCode from '../helpers/statusCode';
+import { respondWithSuccess, respondWithWarning } from '../helpers/responseHandler';
+import { commentEmitter } from '../helpers/notificationHandler';
+import { findTripById, getRequesterEmail } from '../services/tripServices';
+
+
+/**
+ * Create a new Comment
+ * @param {Object} req
+ * @param {Object} res
+ * @returns {Object} comment object
+ */
+export const postComment = async (req, res) => {
+  const data = {
+    tripId: Number(req.params.tripId),
+    userId: req.auth.id,
+    comment: req.body.comment
+  };
+  try {
+    const comment = await createComment(data);
+
+    // get the reason of the trip as title for the notification
+    const reason = await findTripById(Number(req.params.tripId));
+
+    // check if the comment was sent from the manager or the staff
+    const senderRole = (Number(reason.toJSON().userId) !== req.auth.id) ? 'Manager' : null;
```

Comment:
we can destructure `userId` and use it in multiple places where we have ` reason.toJSON().userId`

```
const { userId } = reason.toJSON();
```

Then `Number(userId)`